test/PushNotificationTests/Shared.cpp

```diff
@@ -0,0 +1,155 @@
+            VERIFY_ARE_EQUAL(waitResult, WAIT_OBJECT_0);
+        }
+
+        // Make sure the exitcode for the tool is success.
+        DWORD exitCode{};
+        THROW_IF_WIN32_BOOL_FALSE(GetExitCodeProcess(process.get(), &exitCode));
+        VERIFY_ARE_EQUAL(exitCode, 0);
+    }
+
+    void InstallPackage(const std::wstring& packagePath)
+    {
+        // Deploy packaged app to register handler through the manifest.
+        PackageManager manager;
+        IVector<Uri> depPackagePaths;
+        auto result = manager.AddPackageAsync(Uri(packagePath), depPackagePaths,
+            DeploymentOptions::ForceApplicationShutdown).get();
+        auto errorText = result.ErrorText();
+        auto errorCode = result.ExtendedErrorCode();
+        VERIFY_SUCCEEDED(errorCode, errorText.c_str());
+    }
+
+    void UninstallPackage(const std::wstring& packageFullName)
+    {
+        // Deploy packaged app to register handler through the manifest.
+        PackageManager manager;
+        IVector<Uri> depPackagePaths;
+        auto result = manager.RemovePackageAsync(packageFullName).get();
+        auto errorText = result.ErrorText();
+        auto errorCode = result.ExtendedErrorCode();
+        VERIFY_SUCCEEDED(errorCode, errorText.c_str());
+    }
+
+    wil::unique_event CreateTestEvent(const std::wstring& eventName)
+    {
+        bool alreadyExists = false;
+        SECURITY_ATTRIBUTES attributes = {};
+        wil::unique_hlocal_security_descriptor descriptor;
+
+        // Grant access to world and appcontainer.
+        THROW_IF_WIN32_BOOL_FALSE(ConvertStringSecurityDescriptorToSecurityDescriptor(
+            L"D:(A;;GA;;;WD)(A;;GA;;;AC)", SDDL_REVISION_1, &descriptor, nullptr));
+        attributes.nLength = sizeof(SECURITY_ATTRIBUTES);
+        attributes.lpSecurityDescriptor = descriptor.get();
+
+        wil::unique_event event;
+        event.create(wil::EventOptions::None, eventName.c_str(), &attributes, &alreadyExists);
+        return event;
+    }
+
+    void WaitForEvent(const wil::unique_event& successEvent, const wil::unique_event& failedEvent)
+    {
+        HANDLE waitEvents[2] = { successEvent.get(), failedEvent.get() };
+        auto waitResult = WaitForMultipleObjects(_countof(waitEvents), waitEvents, FALSE,
+            c_phaseTimeout);
+
+        // If waitResult == failureEventIndex, it means the remote test process signaled a
+        // failure event while we were waiting for a different event.
+        auto failureEventIndex = WAIT_OBJECT_0 + 1;
+        VERIFY_ARE_NOT_EQUAL(waitResult, failureEventIndex);
+
+        VERIFY_ARE_NOT_EQUAL(waitResult, WAIT_TIMEOUT);
+        if (waitResult == WAIT_FAILED)
+        {
+            auto lastError = GetLastError();
+            VERIFY_WIN32_FAILED(lastError);
+        }
+
+        successEvent.ResetEvent();
+        failedEvent.ResetEvent();
+    }
+
+    const std::wstring GetDeploymentDir()
```

Comment:
`std::wstring`

You're returning a local/temporary for the caller to do with as they please. No need for it to be const